src/internal/middleware.js

```diff
@@ -91,5 +93,14 @@ export default function sagaMiddlewareFactory(options = {}) {
     return task
   }
 
+  sagaMiddleware.setContext = (props) => {
+    // TODO: or maybe `setContext` should be available on task and this one should act only as proxy method?
+    // would that be useful?
+    // const task = yield takeEvery(...)
+    // const task.setContext({...}) or maybe even - yield setContext(task, {...})?
```

Comment:
hm... at first I thought that this is a valid point and should be done, but after a while i think in the world of code-splitting its nice to have a possibility to share this top-level middleware context among root sagas, as there could be many of those, and when puting this only on task itself it would require such verbose code in order to share it 'globally':
```javascript
const context = {...}
sagaMiddleware.run(...).setContext(context)
// ... later
sagaMiddleware.run(...).setContext(context)
```